### PR TITLE
Tests independent of local import/export config

### PR DESCRIPTION
**Pull request by MioRtia**
#650 

- correction of some Importers methods + added ExportersLoader + test for Importers/Exporters
- making tests independent of configuration for import and export

**NB** Sonar fails because of `com.google.common.base.Supplier` usage and method with too many parameters

### CODE REVIEW

File: afs/afs-ext-base/src/test/java/com/powsybl/afs/ext/base/ImportedCaseTest.java [hunk 1]
```diff
@@ -46,8 +51,12 @@ protected AppStorage createStorage() {
         return MapDbAppStorage.createMem("mem");
     }
 
+    private ExportersLoader createExportersLoader() {
+        return new ExportersLoaderList(Collections.singletonList(new XMLExporter(Mockito.mock(PlatformConfig.class))));
```
Comment (mathbagu) [inline afs/afs-ext-base/src/test/java/com/powsybl/afs/ext/base/ImportedCaseTest.java]: To be fixed if you add the suggested constructor

File: iidm/iidm-converter-api/src/main/java/com/powsybl/iidm/export/Exporters.java [hunk 2]
```diff
@@ -27,18 +28,20 @@
...
+    private static final ExportersLoader LOADER = new ExportersServiceLoader();
+
     private Exporters() {
     }
 
     /**
      * Get all supported export formats.
      */
+    public static Collection<String> getFormats(ExportersLoader loader) {
+        return loader.loadExporters().stream().map(Exporter::getFormat).collect(Collectors.toSet());
```
Comment (mathbagu) [inline iidm/iidm-converter-api/src/main/java/com/powsybl/iidm/export/Exporters.java]: Check if loader is null or not

File: iidm/iidm-converter-api/src/main/java/com/powsybl/iidm/export/Exporters.java [hunk 3]
```diff
@@ -27,18 +28,20 @@
  */
 public final class Exporters {
 
+    private static final ExportersLoader LOADER = new ExportersServiceLoader();
```
Comment (mathbagu) [inline iidm/iidm-converter-api/src/main/java/com/powsybl/iidm/export/Exporters.java]: If ExporterServiceLoader rely on the configuration, you should use a supplier

File: iidm/iidm-converter-api/src/main/java/com/powsybl/iidm/export/ExportersLoaderList.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+/**
+ * @author Miora Ralambotiana <miora.ralambotiana at rte-france.com>
+ */
+public class ExportersLoaderList implements ExportersLoader {
+
+    private final List<Exporter> exporters;
+
+    public ExportersLoaderList(List<Exporter> exporters) {
+        this.exporters = exporters;
```
Comment (mathbagu) [inline iidm/iidm-converter-api/src/main/java/com/powsybl/iidm/export/ExportersLoaderList.java]: Assert that exporters is not null

File: iidm/iidm-converter-api/src/test/java/com/powsybl/iidm/import_/ImportersTest.java [hunk 5]
```diff
@@ -6,41 +6,204 @@
...
+        assertNotNull(importers);
+        assertEquals(1, importers.size());
+        assertTrue(importers.contains(testImporter));
+    }
+
+    @Test
+    public void getImporter() {
+        Importer importer = Importers.getImporter(loader, TEST_FORMAT, computationManager, importConfigMock);
+        assertNotNull(importer);
+        assertEquals(testImporter, importer);
```
Comment (geofjamg) [inline iidm/iidm-converter-api/src/test/java/com/powsybl/iidm/import_/ImportersTest.java]: assertSame?

File: iidm/iidm-converter-api/src/test/java/com/powsybl/iidm/export/ExportersTest.java [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+        Collection<String> formats = Exporters.getFormats(loader);
+        assertEquals(1, formats.size());
+        assertTrue(formats.contains(TEST_FORMAT));
+    }
+
+    @Test
+    public void getExporter() {
+        Exporter exporter = Exporters.getExporter(loader, TEST_FORMAT);
+        assertNotNull(exporter);
+        assertEquals(testExporter, exporter);
```
Comment (geofjamg) [inline iidm/iidm-converter-api/src/test/java/com/powsybl/iidm/export/ExportersTest.java]: assertSame?

File: iidm/iidm-converter-api/src/test/java/com/powsybl/iidm/export/ExportersTest.java [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+    @Test
+    public void export2() throws IOException {
+        Exporters.export(loader, TEST_FORMAT, null, null, path);
+        DataSource dataSource = Exporters.createDataSource(path);
+        assertEquals(Byte.BYTES, dataSource.newInputStream(null, EXTENSION).read());
+    }
+
+    @Test
+    public void export3() throws IOException {
+        Path dir = Files.createTempDirectory("tmp-export");
```
Comment (geofjamg) [inline iidm/iidm-converter-api/src/test/java/com/powsybl/iidm/export/ExportersTest.java]: Files.createTempDirectory does not use jimfs  but create a temp file on the local filesystem. Is it possible to replace by an in memory one?

File: iidm/iidm-converter-api/src/test/java/com/powsybl/iidm/import_/TestImportPostProcessor.java [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+ */
+public class TestImportPostProcessor implements ImportPostProcessor {
+
+    @Override
+    public String getName() {
+        return "test";
+    }
+
+    @Override
+    public void process(Network network, ComputationManager computationManager) throws Exception {
```
Comment (geofjamg) [inline iidm/iidm-converter-api/src/test/java/com/powsybl/iidm/import_/TestImportPostProcessor.java]: throws can be removed

File: iidm/iidm-converter-api/src/test/java/com/powsybl/iidm/export/ExportersTest.java (hunk 7 above)
Comment (MioRtia) [reply iidm/iidm-converter-api/src/test/java/com/powsybl/iidm/export/ExportersTest.java]: The tested method needs the String of the directory name, not a path (the path is deduced by `Paths.get` inside the method), so I don't think so...

**Pull request merged**